plugins/spi/SPIOutput.cpp

```diff
@@ -874,6 +917,239 @@ uint8_t SPIOutput::CalculateAPA102PixelBrightness(uint8_t brightness) {
+    // // BC data could be device dependent to calibrate led colors
+    // uint8_t temp_BCB = 0x7F;  // full
+    // uint8_t temp_BCG = 0x7F;  // full
+    // uint8_t temp_BCR = 0x7F;  // full
+    // device_data.fields.config[1] |=
+    //   (temp_BCB & TLC5971_PACKET_CONFIG_MASKS_BCB)
+    //   >> TLC5971_PACKET_CONFIG_LSHIFT_BCB_RS;
+    // // byte border ------------------------------------------
+    // device_data.fields.config[2] |=
+    //   (temp_BCB & TLC5971_PACKET_CONFIG_MASKS_BCB)
+    //   << TLC5971_PACKET_CONFIG_LSHIFT_BCB_LS;
+    // device_data.fields.config[2] |=
+    //   (temp_BCG & TLC5971_PACKET_CONFIG_MASKS_BCG)
+    //   >> TLC5971_PACKET_CONFIG_LSHIFT_BCG_RS;
+    // // byte border ------------------------------------------
+    // device_data.fields.config[3] |=
+    //   (temp_BCG & TLC5971_PACKET_CONFIG_MASKS_BCG)
+    //   << TLC5971_PACKET_CONFIG_LSHIFT_BCG_LS;
+    // device_data.fields.config[3] |=
+    //   (temp_BCR & TLC5971_PACKET_CONFIG_MASKS_BCR)
+    //   << TLC5971_PACKET_CONFIG_LSHIFT_BCR;
+
+    // // fixed values for testing other things:
+    device_data.fields.config.bytes[0] = 0b10010100;  // 0x94
+    device_data.fields.config.bytes[1] = 0b01011111;  // 0x5F
+    device_data.fields.config.bytes[2] = 0b11111111;  // 0xFF
+    device_data.fields.config.bytes[3] = 0b11111111;  // 0xFF
+
+
+    // OLA_WARN << "FC + BC data:";
+    // for (uint16_t i = 0; i < 4; i++) {
+    //   OLA_WARN << "[" << static_cast<int>(i) << "] "
+    //            << std::bitset<8>(device_data.fields.config.bytes[i]);
+    // }
+
+    // fill gs data
+    // possible with
+    // device_data.gsdata.gs_fields.GSB3 = 65000
+    // or
+    for (
+      uint8_t gs_index = 0;
+      gs_index < TLC5971_SLOTS_PER_DEVICE;
+      gs_index++
+    ) {
+      // OLA_WARN << "  gs_index:"
+      //          << static_cast<uint16_t>(gs_index);
+      // OLA_WARN << "  dmx_offset + gs_index:"
+      //          << static_cast<uint16_t>(dmx_offset + gs_index);
+      device_data.fields.gsdata.bytes[gs_index] =
+        buffer.Get(dmx_offset + gs_index);
+    }
+
+    // OLA_WARN << "GS data:";
+    // for (uint16_t i = 0; i < 24; i++) {
+    //   OLA_WARN << "[" << static_cast<int>(i) << "] "
+    //            << std::bitset<8>(device_data.fields.gsdata.bytes[i]);
+    // }
+
+    // OLA_WARN << "  TLC5971_packet_t size:"
+    //          << sizeof(TLC5971_packet_t);
+    // OLA_WARN << "  device_data size:"
+    //          << sizeof(device_data);
+    // should return 28byte = 224bit
+
+    // copy data to output buffer
+    // memcpy(output + spi_offset, device_data.bytes, sizeof(TLC5971_packet_t));
+    for (
+      uint8_t data_index = 0;
+      data_index < sizeof(TLC5971_packet_t);
+      data_index++
+    ) {
+      output[spi_offset + data_index] = device_data.bytes[data_index];
+    }
+  }  // for devices_in_buffer end
+
+  // write output back
+  m_backend->Commit(m_output_number);
```

Comment:
Low tech debug, add a log line before this, is it this causing the issue, or the code above.